addons/product/models/res_company.py

```diff
@@ -1,12 +1,35 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, _
+from odoo import api, fields, models, _
 
 
 class ResCompany(models.Model):
     _inherit = "res.company"
 
+    def _default_weight_uom_id(self):
+        kg_uom = self.env.ref('product.product_uom_kgm', raise_if_not_found=False)
+        # necessary check as data not yet loaded on module installation
+        if kg_uom:
+            return kg_uom.id
+
+    def _default_volume_uom_id(self):
+        litre_uom = self.env.ref('product.product_uom_litre', raise_if_not_found=False)
+        # necessary check as data not yet loaded on module installation
```

Comment:
yes, sadly the defaults are set before the data is loaded. So we cannot use the default correctly :/

Also explains why we have the <function> tag in the data.

I don't see a better way to do it. (we would need to have it created in a module on which product depends, something like a separate UoM module to do it cleanly...)

This is also why I implemented a fall-back to kg and L if no uom_ids are specified I believe.